### PR TITLE
MCS: 3 sorries in Tcb_R.thy

### PR DESCRIPTION
**Pull request by MitchellBuckley**
Fixes 3 sorries in Tcb_R.thy

### CODE REVIEW

File: proof/refine/ARM/Tcb_R.thy [hunk 1]
```diff
@@ -372,10 +374,33 @@ lemma suspend_ResumeCurrentThread_imp_notct[wp]:
    \<lbrace>\<lambda>rv s. ksSchedulerAction s = ResumeCurrentThread \<longrightarrow> ksCurThread s \<noteq> t'\<rbrace>"
   by (wpsimp simp: suspend_def wp_del: getThreadState_only_state_wp)
 
+(* FIXME RT: move *)
+lemma asUser_valid_tcbs' [wp]:
+  "\<lbrace>valid_tcbs'\<rbrace> asUser t f \<lbrace>\<lambda>rv. valid_tcbs'\<rbrace>"
```
Comment (michaelmcinerney) [inline proof/refine/ARM/Tcb_R.thy:379]: Use invariant notation?

**Review by michaelmcinerney — APPROVED** [review]
Cool!

**Review by mktnk3 — APPROVED** [review]
Nice :)
It would be good if Ryan can have a look through to make sure that these changes fit well with his changes. I’d expect things to be fine at this stage, but just in case.

**Comment by MitchellBuckley** [comment]
Hi @ryybrr ! No rush, but can you have a quick look over this at some point?

**Review by ryybrr — APPROVED** [review]
Looks good to me!

**Comment by MitchellBuckley** [comment]
Thanks all, I'll clean this up and merge.

**Pull request merged**